File: 230-databases-rethinkdb/update.py
import rethinkdb as r
import random

conn = r.connect(host="localhost", port=7704, db="players", auth_key="", timeout=20)

while True:
    scores = r.table("scores")
    id = str(random.randint(1,1000))
    data = scores.get(id).run(conn)
    newScore = random.randint(1,10) + data["Score"]
    r.table("scores").get(id).update({"Score": newScore}).run(conn)

File: 230-databases-rethinkdb/update.go
package main

import (
	"fmt"
	"log"
	"math/rand"
	"strconv"
	"time"

	r "gopkg.in/dancannon/gorethink.v1"
)

//ScoreEntry for scores
// type ScoreEntry struct {
// 	ID         string `gorethink:"id,omitempty"`
// 	PlayerName string
// 	Score      int
// }

func main() {
	fmt.Println("Connecting to RethinkDB")

	session, err := r.Connect(r.ConnectOpts{
		Address:  "localhost:7704",
		Database: "players",
	})

	session.SetMaxOpenConns(10)

	if err != nil {
		log.Fatal("Could not connect")
	}

	fmt.Println("Loop through updates")
	for {
		pl := rand.Intn(1000)
		sc := rand.Intn(6) - 2

		// startingTime := time.Now()
		_, err = r.Table("scores").Get(strconv.Itoa(pl)).Update(map[string]interface{}{
			"Score": r.Row.Field("Score").Add(sc),
		}).RunWrite(session)
		// fmt.Println(time.Now().Sub(startingTime))

		time.Sleep(time.Duration(rand.Int31n(1000)) * time.Millisecond)
	}
}
